src/tag.c

```diff
@@ -1028,6 +1040,15 @@ do_tag(
 	     */
 	    i = jumpto_tag(matches[cur_match], forceit, type != DT_CSCOPE);
 
+	    /*
+	     * Trigger QuickFixCmdPost autocommand.
+	     */
+#if defined(FEAT_QUICKFIX)
+	    if (type == DT_LTAG)
+		apply_autocmds(EVENT_QUICKFIXCMDPOST, (char_u *)"ltag",
```

Comment:
The QuickfixCmdPost autocmd is triggered before jumping to the error.
So this should be moved right after setting the list of tags to the
location list using set_errorlist().

Reply:
That was indeed my first attempt (adding this chunk around the line 928). Unfortunately, and I don't know exactly why, this results in having the buffer to which the first error points to displayed _inside_ the location lists's window. Would you mind helping me a bit with this one ?

Reply:
Sorry in unclear: assuming you have a `QuickFixCmdPost` autocommand to open the location list when non-empty.

Reply:
Actually I've been thinking about that case. I'd say there is a conflict between the meaning of two very different things:

* Jump to a _tag_
* Jump to a quickfix's _location_

As the documentation says:

> :lt[ag][!] [name]	Jump to tag [name] and add the matching tags to a new location list for the current window. [...] The location list showing the matching tags is independent of the tag stack.

As for `QuickFixCmdPost`:

> QuickFixCmdPost			Like QuickFixCmdPre, but after a quickfix 
> 				                command is run, before jumping to the first
>				                location. [...] 

Actually at that point we did jump to the _tag_ as in *pushing to the tag stack*. The location list was filled but we did not explicitely ask for jumping to the first location even if the result is similar.

That's why, instead of trying to change a behavior that could be tricky, I'd suggest considering to explain this particular behavior in the docs. @chrisbra @yegappan please let me know your opinion... :-)